logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -120,9 +146,20 @@ def report(stats)
                 :outputs => plugin_stats(stats, :outputs)
               },
               :reloads => stats[:reloads],
-              :queue => stats[:queue]
+              :queue => extended_stats["queue"]
             }.merge(stats[:dlq] ? {:dead_letter_queue => stats[:dlq]} : {})
+            if !opts[:vertices]
+              ret.delete('vertices')
+              unless extended_stats.nil?
+                extended_stats.delete('vertices')
+              end
+            end
+            if extended_stats.nil?
```

Comment:
`extended_stats` will never be nil here, since we send `Hash#[]` to it in your change on line 149 which would raise a `NoMethodError`.

Reply:
Thanks! Removed.